server/controllers/notification.js

```diff
@@ -0,0 +1,70 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route PUT /profile/edit
+// @desc edit user profile
+// @access Public
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+const userId = await User.findById(req.user.id);
+const { type, title, description } = req.body;
+const notification = await Notification.create({userId, type, title, description});
+
+  res.status(200).json({
+    success: {
+      notification: notification
+    },
+  });
+
+});
+
+
+exports.markAsRead = asyncHandler(async (req, res, next) => {
+const {notificationId} = req.params;
+const update = { read: true };
+const foundNotification = await Notification.findOneAndUpdate(notificationId, update);
+const readNotification = await foundNotification.save()
```

Comment:
Another note about security here:

The `protect` middleware only guarantees the user is authenticated, however,  it doesn't say anything about whether the notification being updated *belongs to* the authenticated user or not. Arguably, we should check for that here and return an appropriate repsonse (e.g. [`403 Forbidden`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)).

Reply:
I used .toString() to convert the object to a string then checking that against req.user.id, and now it should be sending 403 if the ids don't match.